src/scenes/characters/character.gd

```diff
@@ -17,22 +17,39 @@ func hover_unfocus() -> void:
 func interact() -> void:
     if (!GameState.is_in_interaction()):
         GameState.set_is_in_dialogue(true);
-        check_quest_status();
+        var quest_in_progress = check_quest_status();
+        if (quest_in_progress):
+            GameState.set_is_in_dialogue(false);
+            return;
         var index = Dialogic.get_variable(npc + '_index');
         var dialog = Dialogic.start(npc + str(index));
         dialog.connect("timeline_end", self, "dialog_ended");
         dialog.connect("dialogic_signal", self, "dialogic_signal_event");
         add_child(dialog);
 
-func check_quest_status() -> void:
+func check_quest_status() -> bool:
     for quest in Quests.quest_list:
         if quest.label == fetch_quest.quest.label:
             for item in Inventory.held_items:
                 if item.label == fetch_quest.quest.item.label:
                     Dialogic.set_variable(npc + '_quest_complete', true);
+                    return false;
+            return true;
+    return false;
 
-func dialog_ended(_timeline_name) -> void:
+func dialog_ended(timeline_name) -> void:
     GameState.set_is_in_dialogue(false);
+    var timeline_name_length = timeline_name.length();
+    var timeline_number = int(timeline_name[timeline_name_length - 1]);
+    if (timeline_number == 2):
+        # If timeline_number == 2, no new hints are provided in the corresponding dialog
```

Comment:
It would be better to assign the value 2 to a named constant whose name indicates the special condition, e.g.:
```gdscript
const NO_HINT_TIMELINE = 2;

func dialog_ended(timeline_name) -> void:
  # ...
  if (timeline_number != NO_HINT_TIMELINE):
    var character_name = timeline.name.left(timeline_name_length - 1);
    add_hint(character_name);
```
If you can come up with a better name for the constant, that would be better (the context is not quite clear to me).
Btw, in this case instead of early exit it is better to directly indicate the logic branch condition.

Reply:
The constant name looks good to me. Makes sense, I've updated the `if` statement to directly indicate the logic branch condition.